interceptor/interceptor2.go

```diff
@@ -0,0 +1,41 @@
+package interceptor
+
+import (
+	"context"
+	"io"
+
+	"github.com/pion/rtcp"
+	"github.com/pion/rtp"
+	"github.com/pion/webrtc/v3"
+)
+
+type Interceptor2 interface {
+	Intercept(context.Context, *webrtc.PeerConnection, ReadWriter) ReadWriter
+}
+
+// Reader is an interface to handle incoming RTP stream.
+type ReadWriter interface {
+	ReadRTP(context.Context) (*rtp.Packet, map[interface{}]interface{}, error)
```

Comment:
I would avoid using `map[interface{}]interface{}` if possible.

I need to look at HTTP libraries more, but it would be possible to use `context.Context`? I worry that people will see this and judge Pion unfairly. `interface{}` in a public API is something people usually push back on.

Reply:
I added map[interface{}]interface{} because context seemed confusing; there is a context for the read/write operation (that can be used for the context timeout/cancel, i guess), and there is a context that would be used to pass packet metadata:

```
	ReadRTP(ctx context.Context) (packet *rtp.Packet, packetContext context.Context, err error)
	WriteRTP(ctx context.Context, packet *rtp.Packet, packetContext context.Context) error
```

At the end, context.WithValue(ctx, key, val) uses interface{} as well for keys and values. We could use string keys as well, but then there could be conflict between the different interceptors. 

Here is how i imagined using meta (and the same with the recomended context usage):
```
type dummyKey struct{}

func GetDummy(meta map[interface{}]interface{}) string {
	return meta[dummyKey{}].(string)
}

func setDummy(meta map[interface{}]interface{}, val string) {
	meta[dummyKey{}] = val
}

func GetDummyCtx(ctx context.Context) string {
	return ctx.Value(dummyKey{}).(string)
}

func setDummyCtx(ctx context.Context, val string) context.Context {
	return context.WithValue(ctx, dummyKey{}, val)
}
```

Alternatively we could hide the map[interface{}]interface{} with something like:
```
type RTPPacketWithMeta struct {
	rtp.Packet
	meta map[interface{}]interface{}
}

func (p *RTPPacketWithMeta) GetMeta(key interface{}) (interface{}, bool) {
	val, ok := p.meta[key]
	return val, ok
}

func (p *RTPPacketWithMeta) SetMeta(key, val interface{}) {
	p.meta[key] = val
}
```

But i'm open to other suggestions as well, i just feel like using two context.Context in one method would be confusing.